### DISCUSSION TITLE
Don't log 404s to console

### DISCUSSION THREAD

**Discussion by Dig-Doug** [sveltejs/kit#12900] [Q&A] [answered]
Is there a way to catch 404 errors so I can 1) prevent them from being logged in the console and 2) count them for alerting?

I want to stop them from being logged because my server's logs get analyzed for exceptions, e.g.

```
SvelteKitError: Not found: /web/wp-includes/wlwmanifest.xml
at .resolve2 ( file:///site/build/server/index.js:3414 )
at .resolve ( file:///site/build/server/index.js:3247 )
at .resolve ( file:///site/build/server/chunks/hooks.server-B0Tx0_fk.js:37 )
```

**Comment by EskelCz** [comment, marked as answer]
The only way I found is this:
hooks.server.js
```
export function handleError({ error }) {
  if (!error.message.includes('Not found')) { // Ignore 404 Not Found
    console.error(error.stack)
    return { message: 'Internal Error' }
  }
}
```

**Comment by simonsarris** [comment]
Is there no more canonical way than this, @Rich-Harris ? This could really blow up log size on some sites

**Reply by Rich-Harris** [reply]
Feels like the stack trace is the unnecessary part and we should just omit it for 404s. I think it would be weird to omit the 404s themselves, that feels like something you'd want to know about

**Reply by Rich-Harris** [reply]
Actually... it appears this is already the case in more recent versions
